src/Functions/keyvaluepair/extractKeyValuePairs.cpp

```diff
@@ -0,0 +1,213 @@
+    }
+
+    if (!parsed_arguments.pair_delimiters.empty())
+    {
+        builder.withItemDelimiters(parsed_arguments.pair_delimiters);
+    }
+
+    if (parsed_arguments.quoting_character)
+    {
+        builder.withQuotingCharacter(parsed_arguments.quoting_character.value());
+    }
+
+    return builder.build();
+}
+
+ColumnPtr extract(ColumnPtr data_column, std::shared_ptr<KeyValuePairExtractor> extractor)
+{
+    auto offsets = ColumnUInt64::create();
+
+    auto keys = ColumnString::create();
+    auto values = ColumnString::create();
+
+    uint64_t offset = 0u;
+
+    for (auto i = 0u; i < data_column->size(); i++)
+    {
+        auto row = data_column->getDataAt(i).toView();
+
+        auto pairs_count = extractor->extract(row, keys, values);
+
+        offset += pairs_count;
+
+        offsets->insert(offset);
+    }
+
+    keys->validate();
+    values->validate();
+
+    ColumnPtr keys_ptr = std::move(keys);
+
+    return ColumnMap::create(keys_ptr, std::move(values), std::move(offsets));
+}
+
+
+}
+
+namespace DB
+{
+
+String ExtractKeyValuePairs::getName() const
+{
+    return name;
+}
+
+FunctionPtr ExtractKeyValuePairs::create(ContextPtr)
+{
+    return std::make_shared<ExtractKeyValuePairs>();
+}
+
+DataTypePtr ExtractKeyValuePairs::getReturnTypeImpl(const DataTypes &) const
+{
+    return std::make_shared<DataTypeMap>(std::make_shared<DataTypeString>(), std::make_shared<DataTypeString>());
+}
+
+bool ExtractKeyValuePairs::isVariadic() const
+{
+    return true;
+}
+
+bool ExtractKeyValuePairs::isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo &) const
+{
+    return false;
+}
+
+std::size_t ExtractKeyValuePairs::getNumberOfArguments() const
+{
+    return 0u;
+}
+
+ColumnNumbers ExtractKeyValuePairs::getArgumentsThatAreAlwaysConstant() const
+{
+    return {1, 2, 3, 4, 5};
+}
+
+ColumnPtr ExtractKeyValuePairs::executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t) const
+{
+    auto parsed_arguments = ArgumentExtractor::extract(arguments);
+
+    auto extractor_without_escaping = getExtractor(parsed_arguments);
+
+    return extract(parsed_arguments.data_column, extractor_without_escaping);
+}
+
+REGISTER_FUNCTION(ExtractKeyValuePairs)
+{
+    factory.registerFunction<ExtractKeyValuePairs>(
+        Documentation(
+            R"(Extracts key-value pairs from any string. The string does not need to be 100% structured in a key value pair format;
+
+            It can contain noise (e.g. log files). The key-value pair format to be interpreted should be specified via function arguments.
+
+            A key-value pair consists of a key followed by a `key_value_delimiter` and a value. Quoted keys and values are also supported. Key value pairs must be separated by pair delimiters.
+            Escaping support can be turned on and off.
+
+            Escape sequences supported: `\x`, `\N`, `\a`, `\b`, `\e`, `\f`, `\n`, `\r`, `\t`, `\v` and `\0`.
```

Comment:
Does it really still supports octal numerals? **\`\0\`** or is it just an outdated comment?

Reply:
Null-terminator character